File: AndroidLab/ch6/src/main/java/com/example/ch6/section5/Test5_1Activity.java
package com.example.ch6.section5;

import android.os.Bundle;
import android.os.SystemClock;
import android.widget.SeekBar;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.example.ch6.R;
import com.example.ch6.databinding.ActivityTest51Binding;

public class Test5_1Activity extends AppCompatActivity {

    ActivityTest51Binding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);

        binding = ActivityTest51Binding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        //thread 구동
        ProgressThread thread = new ProgressThread();
        thread.start();

        binding.seek.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            //값이 변경되는 순간마다..
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                binding.resultView.setText("현재 값 : "+progress);
            }
            //값 조정을 위해 터치하는 순간..
            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }
            //터치를 떼는 순간..
            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
                Toast.makeText(Test5_1Activity.this, "선택한 값 : "+seekBar.getProgress(),
                        Toast.LENGTH_SHORT).show();
            }
        });
    }

    class ProgressThread extends Thread {
        @Override
        public void run() {
            for(int i = 0; i < 10; i++){
                //1초씩 쉬었다가...
                SystemClock.sleep(1000);
                binding.progress.incrementProgressBy(10);
                binding.progress.incrementSecondaryProgressBy(15);
            }
        }
    }
}

File: AndroidLab2/ch1/src/main/java/com/example/ch1/section5_fab_snackbar/Test5_1Activity.kt
package com.example.ch1.section5_fab_snackbar

import android.os.Bundle
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.ch1.R
import com.example.ch1.databinding.ActivityTest51Binding
import com.google.android.material.snackbar.Snackbar

class Test5_1Activity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        val binding = ActivityTest51Binding.inflate(layoutInflater)
        setContentView(binding.root)

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        binding.fab.shrink()//초기상태.. icon 만 나오게..

        binding.fab.setOnClickListener {
            when(binding.fab.isExtended){
                true -> {
                    Snackbar.make(binding.main, "메시지를 확인하시겠습니까?", Snackbar.LENGTH_LONG)
                        .setAction("실행취소"){
                            //액션 이벤트....
                            Toast.makeText(this@Test5_1Activity, "메시지 확인", Toast.LENGTH_SHORT)
                                .show()
                        }
                        .show()
                    binding.fab.shrink()
                }
                false -> binding.fab.extend()
            }
        }
    }
}